docs/design/pr-builder.rst

```diff
@@ -0,0 +1,103 @@
+Design of Pull Request Builder
+==============================
+
+Background
+----------
+
+This will focus on automatically building documentation for Pull Requests
+on Read the Docs projects. This is one of the most requested feature of
+Read the Docs. This document will serve as a design document
+for discussing how to implement this features.
+
+Scope
+-----
+
+- Receiving ``pull_request`` webhook event from Github
+- Fetching data from pull requests.
+- Making Pull Requests work like temporary ``Version``
+- Creating PR Versions when a pull request is opened and Triggering a build
+- Setting PR version Life time
+- Storing PR Version build Data.
+- Deleting PR version and the build data
+- Excluding PR Versions from Search Engines
+- Serving PR Docs
```

Comment:
Probably we should add reporting to GitHub too? Like commit status or just a link to the build docs

Reply:
Good Idea Thanks :100: